[CI] TransportAddVotingConfigExclusionsActionTests testExcludeExistingNodesByNodeNames failure

Failed on master: https://gradle-enterprise.elastic.co/s/z5oe3y3z3iovg

Does _not_ reproduce locally with:

```
./gradlew ':server:test' --tests "org.elasticsearch.action.admin.cluster.configuration.TransportAddVotingConfigExclusionsActionTests.testExcludeExistingNodesByNodeNames" -Dtests.seed=5DC87794C671DDFD -Dtests.security.manager=true -Dtests.locale=nl-NL -Dtests.timezone=America/Cuiaba -Dcompiler.java=14
```

Error:

```
Captured an uncaught exception in thread: Thread[id=37, name=elasticsearch[test][masterService#updateTask][T#1], state=RUNNABLE, group=TGRP-TransportAddVotingConfigExclusionsActionTests]
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=37, name=elasticsearch[test][masterService#updateTask][T#1], state=RUNNABLE, group=TGRP-TransportAddVotingConfigExclusionsActionTests]Open stacktrace
Caused by: java.lang.AssertionError: 

Expected: sameInstance(<cluster uuid: _na_ [committed: false]
version: 2
state uuid: BIJz_KXpR4WjWyr1JWUNOQ
from_diff: false
meta data version: 1
   coordination_metadata:
      term: 0
      last_committed_config: VotingConfiguration{other2,other1,local}
      last_accepted_config: VotingConfiguration{other2,other1,local}
      voting tombstones: [{_absent_}{absent_id}]
metadata customs:
   index-graveyard: IndexGraveyard[[]]
nodes: 
   {other1}{other1}{R3ct-fgOQmaj5ewaVb68Hw}{0.0.0.0}{0.0.0.0:2}{m}
   {other2}{other2}{-dZ5hvsoRZqTP0NBGA4Zxg}{0.0.0.0}{0.0.0.0:3}{m}
   {local}{local}{23QUFO0vQvOh2NNmw3Ik-A}{0.0.0.0}{0.0.0.0:1}{m}, local, master
   {data}{data}{eQv0CveFTj6yLyHYRVe4MA}{0.0.0.0}{0.0.0.0:4}
routing_table (version 0):
routing_nodes:
---- unassigned
>)
     but: was <cluster uuid: _na_ [committed: false]
version: 2
state uuid: E5m2s_08TAyudx3n1nF6Uw
from_diff: false
meta data version: 1
   coordination_metadata:
      term: 0
      last_committed_config: VotingConfiguration{other2,other1,local}
      last_accepted_config: VotingConfiguration{other2,other1,local}
      voting tombstones: [{other1}{other1}, {other2}{other2}]
metadata customs:
   index-graveyard: IndexGraveyard[[]]
nodes: 
   {other1}{other1}{R3ct-fgOQmaj5ewaVb68Hw}{0.0.0.0}{0.0.0.0:2}{m}
   {other2}{other2}{-dZ5hvsoRZqTP0NBGA4Zxg}{0.0.0.0}{0.0.0.0:3}{m}
   {local}{local}{23QUFO0vQvOh2NNmw3Ik-A}{0.0.0.0}{0.0.0.0:1}{m}, local, master
   {data}{data}{eQv0CveFTj6yLyHYRVe4MA}{0.0.0.0}{0.0.0.0:4}
routing_table (version 0):
routing_nodes:
---- unassigned
>
Open stacktrace
[2020-04-20T14:02:18,326][INFO ][o.e.a.a.c.c.TransportAddVotingConfigExclusionsActionTests] [testExcludeExistingNodesByNodeNames] before test
[2020-04-20T14:02:18,409][INFO ][o.e.a.a.c.c.TransportAddVotingConfigExclusionsActionTests] [testExcludeExistingNodesByNodeN
```

--- a/server/src/test/java/org/elasticsearch/action/admin/cluster/configuration/TransportAddVotingConfigExclusionsActionTests.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/cluster/configuration/TransportAddVotingConfigExclusionsActionTests.java
@@ -149,9 +149,9 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
     }
 
     public void testWithdrawsVoteFromANode() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
 
-        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions());
+        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions(countDownLatch));
         transportService.sendRequest(localNode, AddVotingConfigExclusionsAction.NAME, new AddVotingConfigExclusionsRequest("other1"),
             expectSuccess(r -> {
                 assertNotNull(r);
@@ -164,9 +164,9 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
     }
 
     public void testWithdrawsVotesFromMultipleNodes() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
 
-        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions());
+        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions(countDownLatch));
         transportService.sendRequest(localNode, AddVotingConfigExclusionsAction.NAME,
             new AddVotingConfigExclusionsRequest("other1", "other2"),
             expectSuccess(r -> {
@@ -181,9 +181,9 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
     }
 
     public void testWithdrawsVotesFromNodesMatchingWildcard() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
 
-        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions());
+        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions(countDownLatch));
         transportService.sendRequest(localNode, AddVotingConfigExclusionsAction.NAME, makeRequestWithNodeDescriptions("other*"),
             expectSuccess(r -> {
                 assertNotNull(r);
@@ -198,9 +198,9 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
     }
 
     public void testWithdrawsVotesFromAllMasterEligibleNodes() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
 
-        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions());
+        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions(countDownLatch));
         transportService.sendRequest(localNode, AddVotingConfigExclusionsAction.NAME, makeRequestWithNodeDescriptions("_all"),
             expectSuccess(r -> {
                 assertNotNull(r);
@@ -215,9 +215,9 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
     }
 
     public void testWithdrawsVoteFromLocalNode() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
 
-        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions());
+        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions(countDownLatch));
         transportService.sendRequest(localNode, AddVotingConfigExclusionsAction.NAME, makeRequestWithNodeDescriptions("_local"),
             expectSuccess(r -> {
                 assertNotNull(r);
@@ -295,9 +295,9 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
     }
 
     public void testExcludeAbsentNodesByNodeIds() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
 
-        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions());
+        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions(countDownLatch));
         transportService.sendRequest(localNode, AddVotingConfigExclusionsAction.NAME,
             new AddVotingConfigExclusionsRequest(Strings.EMPTY_ARRAY, new String[]{"absent_id"},
                                                     Strings.EMPTY_ARRAY, TimeValue.timeValueSeconds(30)),
@@ -312,9 +312,9 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
     }
 
     public void testExcludeExistingNodesByNodeIds() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
 
-        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions());
+        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions(countDownLatch));
         transportService.sendRequest(localNode, AddVotingConfigExclusionsAction.NAME,
             new AddVotingConfigExclusionsRequest(Strings.EMPTY_ARRAY, new String[]{"other1", "other2"},
                                                     Strings.EMPTY_ARRAY, TimeValue.timeValueSeconds(30)),
@@ -330,9 +330,9 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
     }
 
     public void testExcludeAbsentNodesByNodeNames() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
 
-        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions());
+        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions(countDownLatch));
         transportService.sendRequest(localNode, AddVotingConfigExclusionsAction.NAME, new AddVotingConfigExclusionsRequest("absent_node"),
             expectSuccess(e -> {
                 countDownLatch.countDown();
@@ -345,9 +345,9 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
     }
 
     public void testExcludeExistingNodesByNodeNames() throws InterruptedException {
-        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        final CountDownLatch countDownLatch = new CountDownLatch(2);
 
-        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions());
+        clusterStateObserver.waitForNextChange(new AdjustConfigurationForExclusions(countDownLatch));
         transportService.sendRequest(localNode, AddVotingConfigExclusionsAction.NAME,
             new AddVotingConfigExclusionsRequest("other1", "other2"),
             expectSuccess(r -> {
@@ -548,7 +548,14 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
         };
     }
 
-    private class AdjustConfigurationForExclusions implements Listener {
+    private static class AdjustConfigurationForExclusions implements Listener {
+
+        final CountDownLatch doneLatch;
+
+        AdjustConfigurationForExclusions(CountDownLatch latch) {
+            this.doneLatch = latch;
+        }
+
         @Override
         public void onNewClusterState(ClusterState state) {
             clusterService.getMasterService().submitStateUpdateTask("reconfiguration", new ClusterStateUpdateTask() {
@@ -571,6 +578,11 @@ public class TransportAddVotingConfigExclusionsActionTests extends ESTestCase {
                 @Override
                 public void onFailure(String source, Exception e) {
                     throw new AssertionError("unexpected failure", e);
+                }
+
+                @Override
+                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                    doneLatch.countDown();
                 }
             });
         }

